src/managers/PunishmentManager.ts

```diff
@@ -192,6 +193,7 @@ interface IAdditionalPunishmentParams {
 const AUTOMATIC: string = "Automatic";
 
 export namespace PunishmentManager {
+    const _logger = new Logger(__filename, false);
```

Comment:
Maybe we can just have one `Logger` for the entire file as opposed to each namespace. If you do that, change the name to `Logger` or `LOGGER`